osu.Game.Rulesets.Catch/Edit/Blueprints/JuiceStreamSelectionBlueprint.cs

```diff
@@ -26,13 +35,26 @@ public class JuiceStreamSelectionBlueprint : CatchSelectionBlueprint<JuiceStream
 
         private readonly Cached pathCache = new Cached();
 
+        private readonly SelectionEditablePath editablePath;
+
+        private int lastEditablePathId = -1;
+
+        private int lastSliderPathVersion = -1;
```

Comment:
These are a bit difficult to follow, and probably need renaming and/or xmldoc. I presume the first is the editor's path, and the second is the beatmap's (as encoded in the legacy format?)

There's also `editablePath.PathId` which I'm not entirely sure how it differs from `lastEditablePathId` (is the latter the last editor path ID that got encoded to the beatmap?)